Recommend_Function.R

```diff
@@ -23,18 +25,22 @@ next_module <- function (module="pred", path_Start="pred")
                            tc_pca[["x"]][111:132,1])
   colnames(combined_matrix) <- c("pred","nlp","sna","neural","viz","loop")
   rownames(combined_matrix) <- i1names
-  t_comb_matrix <- t(combined_matrix)
+  #loop through each of the potential modules, creating one column per module as the start point
   unit.order <- matrix(nrow=6,ncol=6)
   for (j in 1:dim(combined_matrix)[2]){
+  #peform cosine similariry and remove self-referential comparisons
     C.SIM <- cosine(combined_matrix)
     diag(C.SIM) <- NA
     last.unit <- colnames(combined_matrix)[j]
+    #loop through cosine similarity matrix finding the next unit, and then clearing that row, so 
+    #we can resort without that row coming up again.  Build path for this starting unit.
     for (i in 1:dim(C.SIM)[2]) {
```

Comment:
You could look at `lapply` to do these kinds of functions, when serious for looping is needed usually someone writes a C version that is run inside R. That's what the apply functions do.